// testing-library/fesm5/testing-library.js
import { Injectable, Component, NgModule, defineInjectable } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
var TestingLibraryService = /** @class */ (function () {
    function TestingLibraryService() {
    }
    TestingLibraryService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] },
    ];
    /** @nocollapse */
    TestingLibraryService.ctorParameters = function () { return []; };
    /** @nocollapse */ TestingLibraryService.ngInjectableDef = defineInjectable({ factory: function TestingLibraryService_Factory() { return new TestingLibraryService(); }, token: TestingLibraryService, providedIn: "root" });
    return TestingLibraryService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
var TestingLibraryComponent = /** @class */ (function () {
    function TestingLibraryComponent() {
    }
    /**
     * @return {?}
     */
    TestingLibraryComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    TestingLibraryComponent.decorators = [
        { type: Component, args: [{
                    selector: 'NgxMat-TestingLibrary',
                    template: "\n    <p>\n      testing-library works!\n    </p>\n  ",
                    styles: []
                },] },
    ];
    /** @nocollapse */
    TestingLibraryComponent.ctorParameters = function () { return []; };
    return TestingLibraryComponent;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
var TestingLibraryModule = /** @class */ (function () {
    function TestingLibraryModule() {
    }
    TestingLibraryModule.decorators = [
        { type: NgModule, args: [{
                    imports: [],
                    declarations: [TestingLibraryComponent],
                    exports: [TestingLibraryComponent]
                },] },
    ];
    return TestingLibraryModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */

export { TestingLibraryService, TestingLibraryComponent, TestingLibraryModule };
