// testing-library/bundles/testing-library.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core')) :
    typeof define === 'function' && define.amd ? define('testing-library', ['exports', '@angular/core'], factory) :
    (factory((global['testing-library'] = {}),global.ng.core));
}(this, (function (exports,i0) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */
    var TestingLibraryService = (function () {
        function TestingLibraryService() {
        }
        TestingLibraryService.decorators = [
            { type: i0.Injectable, args: [{
                        providedIn: 'root'
                    },] },
        ];
        /** @nocollapse */
        TestingLibraryService.ctorParameters = function () { return []; };
        /** @nocollapse */ TestingLibraryService.ngInjectableDef = i0.defineInjectable({ factory: function TestingLibraryService_Factory() { return new TestingLibraryService(); }, token: TestingLibraryService, providedIn: "root" });
        return TestingLibraryService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */
    var TestingLibraryComponent = (function () {
        function TestingLibraryComponent() {
        }
        /**
         * @return {?}
         */
        TestingLibraryComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        TestingLibraryComponent.decorators = [
            { type: i0.Component, args: [{
                        selector: 'NgxMat-TestingLibrary',
                        template: "\n    <p>\n      testing-library works!\n    </p>\n  ",
                        styles: []
                    },] },
        ];
        /** @nocollapse */
        TestingLibraryComponent.ctorParameters = function () { return []; };
        return TestingLibraryComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */
    var TestingLibraryModule = (function () {
        function TestingLibraryModule() {
        }
        TestingLibraryModule.decorators = [
            { type: i0.NgModule, args: [{
                        imports: [],
                        declarations: [TestingLibraryComponent],
                        exports: [TestingLibraryComponent]
                    },] },
        ];
        return TestingLibraryModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */

    exports.TestingLibraryService = TestingLibraryService;
    exports.TestingLibraryComponent = TestingLibraryComponent;
    exports.TestingLibraryModule = TestingLibraryModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
