// testing-library/fesm2015/testing-library.js
import { Injectable, Component, NgModule, defineInjectable } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
class TestingLibraryService {
    constructor() { }
}
TestingLibraryService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] },
];
/** @nocollapse */
TestingLibraryService.ctorParameters = () => [];
/** @nocollapse */ TestingLibraryService.ngInjectableDef = defineInjectable({ factory: function TestingLibraryService_Factory() { return new TestingLibraryService(); }, token: TestingLibraryService, providedIn: "root" });

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
class TestingLibraryComponent {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
TestingLibraryComponent.decorators = [
    { type: Component, args: [{
                selector: 'NgxMat-TestingLibrary',
                template: `
    <p>
      testing-library works!
    </p>
  `,
                styles: []
            },] },
];
/** @nocollapse */
TestingLibraryComponent.ctorParameters = () => [];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
class TestingLibraryModule {
}
TestingLibraryModule.decorators = [
    { type: NgModule, args: [{
                imports: [],
                declarations: [TestingLibraryComponent],
                exports: [TestingLibraryComponent]
            },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */

export { TestingLibraryService, TestingLibraryComponent, TestingLibraryModule };
